django_celery_beat/signals.py

```diff
@@ -0,0 +1,41 @@
+def signals_connect():
+    from django.db.models import signals, OneToOneRel
```

Comment:
OneToOneRel is a private API as far as I know, can you provide a testt case of examples of what you were actually trying to acheive?

Reply:
Test case:

```python3
# admin.py
class MyPeriodicTaskInline(admin.StackedInline):
    model = MyPeriodicTask
    # ...
    # I want to render a `StackedInline` of `MyPeriodicTask` in `OtherModelAdmin`.

class OtherModelAdmin(admin.ModelAdmin):
    model = OtherModel
    inlines = (
        MyPeriodicTaskInline,
    )
```
```python3
# models.py
class MyPeriodicTask(PeriodicTask):
    other_model = models.ForeignKey(
        'OtherModel',
        null=True,
        on_delete=models.CASCADE
    )
    def save(self, *args, **kwargs):
        self.kwargs = json.dumps('a_field_key',self.other_model.a_field)
        super().save(*args, **kwargs)
        # The bug is in it: 
        # This `save()` will call the `save_base()`.
        # But in `save_base()`, it will set `pre_save.sender = MyPeriodicTask`, not `PeriodicTask`.
        # So, the `PeriodicTask` can not receive signal with
        # [signals.pre_delete.connect(PeriodicTasks.changed, sender=PeriodicTask)](https://github.com/celery/django-celery-beat/blob/d4b97af62889527582194f3932e013ad972a88b3/django_celery_beat/models.py#L625)
        # [signals.pre_save.connect(PeriodicTasks.changed, sender=PeriodicTask)](https://github.com/celery/django-celery-beat/blob/d4b97af62889527582194f3932e013ad972a88b3/django_celery_beat/models.py#L625)
```
Even without not overwrite `save()`, the `pre_save.sender` still be `MyPeriodicTask.__class__`.
Because in `MyPeriodicTask`, the `self` is `MyPeriodicTask`.
Amd Django use [self.__class__](https://github.com/django/django/blob/e9fd2b572410b1236da0d3d0933014138d89f44e/django/db/models/base.py#L860) to set `sender`.
`delete()` is the same, look [this](https://github.com/django/django/blob/e9fd2b572410b1236da0d3d0933014138d89f44e/django/db/models/deletion.py#L433).

Reply:
We have two way fix it.
1. Don't use signal, call `PeriodicTasks.changed()` in `PeriodicTask.save()` and `PeriodicTask.delete()`.
2. Like this PR, use reverse related to get all `OneToOneRel`, then get thier `__class__`, connect to the signal.

Reply:
i would prefer first approach

Reply:
can you also add the tests as well

Reply:
> can you also add the tests as well

Yep, but I got the problem like this:
https://stackoverflow.com/questions/502916/django-how-to-create-a-model-dynamically-just-for-testing

Reply:
is that related to this patch?

Reply:
> 



> is that related to this patch?

Yep, I solved it.
Because I need to create the model in testing.